python/ray/data/datasource/parquet_datasource.py

```diff
@@ -84,9 +87,13 @@ def read_pieces(serialized_pieces: List[str]) -> pa.Table:
 
             from pyarrow.dataset import _get_partition_keys
 
+            ctx = DatasetContext.get_current()
+            output_buffer = BlockOutputBuffer(
+                block_udf=_block_udf,
+                target_max_block_size=ctx.target_max_block_size)
+
             logger.debug(f"Reading {len(pieces)} parquet pieces")
             use_threads = reader_args.pop("use_threads", False)
-            tables = []
             for piece in pieces:
                 table = piece.to_table(
```

Comment:
This reads a full fragment (file) at a time, so this should be switched to a streaming read. Maybe using the `Fragment.scanner()` API? https://github.com/apache/arrow/blob/30ac30fd0e375bae1e57b99e38b782e634773fbe/python/pyarrow/_dataset.pyx#L987-L1029

Reply:
Done.